lib/matplotlib/tests/test_colors.py

```diff
@@ -44,6 +45,19 @@ def test_resample():
     assert_array_almost_equal(lc3([0, 0.5, 1]), expected)
 
 
+def test_colormap_copy():
+    cm = plt.cm.Reds
+    cm([-1, 0, .5, np.nan, np.inf])
+    cm.set_bad('y')
+    cm2 = copy.copy(cm)
+    cm2.set_bad('b')
+    cm.set_under('k')
+    cm2.set_under('r')
+    cm.set_over('c')
+    cm2.set_over('m')
+    assert_equal(np.array_equal(cm._lut, cm2._lut), False)
```

Comment:
I am not a huge fan of this test, I would prefer 

```python
assert_array_equal(cm(data), expected1)
assert_array_equal(cm2(data), expected2)
```

As that is the thing thing we actually care about (not the details of the internal state).  If we change how the color mapping works internally, this test might not cover what we think it covers.